src/setfit/onnx.py

```diff
@@ -0,0 +1,227 @@
+
+        hidden = self.pooler(hidden)["sentence_embedding"]
+        if self.model_head is None:
+            return hidden
+
+        out = self.model_head(hidden)
+        return out
+
+
+def sklearn_head_to_onnx(model_head: LogisticRegression, opset: int):
+    """
+    Convert the sklearn head from a SetFitModel to ONNX format.
+
+    :param model_head: The trained SetFit model_head.
+    :param opset: The ONNX opset to use for optimizing this model. The opset is not
+        gauranteed and will default to the maximum version possible for the sklearn
+        model.
+    :raises ImportError: If skl2onnx is not installed an error will be raised asking
+        to install this package.
+    :return: The ONNX model generated from the sklearn head.
+    """
+
+    # Check if skl2onnx is installed
+    try:
+        from skl2onnx.common.data_types import guess_data_type
+        from skl2onnx import convert_sklearn
+        import onnxconverter_common
+        from skl2onnx.sklapi import CastTransformer
+        from sklearn.pipeline import Pipeline
+    except ImportError:
+        msg = (
+            "skl2onnx must be installed in order to convert a model with an sklearn head."
+        )
+        raise ImportError(msg)
+
+    # Check to see that the head has a coef_
+    if not hasattr(model_head, "coef_"):
+        raise ValueError(
+            "Head must have coef_ attribute check that this is supported by your model and the model has been fit."
+        )
+
+    # Determine the initial type and the shape of the output.
+    input_shape = (None, *model_head.coef_.shape[1:])
+    dtype = guess_data_type(model_head.coef_, shape=input_shape)[0][1]
+    dtype.shape = input_shape
+
+    # If the datatype of the model is double we need to cast the outputs
+    # from the setfit model to doubles for compatibility inside of ONNX.
+    if type(dtype) == onnxconverter_common.data_types.DoubleTensorType:
+        # TODO:: TALK ABOUT FLOAT CONVERSION ISSUES.
+        sk_model = Pipeline(
+            [("castdouble", CastTransformer(dtype=np.double)), ("head", model_head)]
+        )
+    else:
+        sk_model = model_head
+
+    # Convert sklearn head into ONNX format
+    initial_type = [("model_head", dtype)]
+    onx = convert_sklearn(
+        sk_model,
+        initial_types=initial_type,
+        target_opset=opset,
+        options={id(sk_model): {"zipmap": False}},
+    )
+
+    return onx
+
+
+def export_onnx(
+    model_body_sentence_transformer,
+    model_head,
+    opset: int,
+    output: str,
+    ignore_ir_version: bool = True,
+):
+    """
+    Export a PyTorch backed setfit model to ONNX Intermediate Representation.
+
+    Args:
+        model_body_sentence_transformer: The model_body from a SetFitModel.model_body. This should be a
+            SentenceTransformer.
+        model_head: The model_head from SetFitModel.model_head. This can be either a dense
```

Comment:
same comment as above:
```suggestion
        model_head: The SetFit model head. This can be either a dense
```